client/extensions/zoninator/state/zones/reducer.js

```diff
@@ -17,6 +23,17 @@ export const requesting = createReducer( {}, {
 
 export const items = createReducer( {}, {
 	[ ZONINATOR_UPDATE_ZONES ]: ( state, { siteId, data } ) => ( { ...state, [ siteId ]: data } ),
+	[ ZONINATOR_UPDATE_ZONE ]: ( state, { siteId, data } ) => {
+		const zones = [ ...( state[ siteId ] || [] ) ];
+		const idx = findIndex( zones, { term_id: data.term_id || -1 } );
+
+		if ( idx < 0 ) {
+			return { ...state, [ siteId ]: zones.concat( [ data ] ) };
```

Comment:
In this case `data` stands for a single zone, while `state[ sideId ]` is an array of zones.  
Though now I'm wondering why I didn't just use `zones.push( data )`.